crates/shared/src/sources/balancer_v2/pools/common.rs

```diff
@@ -189,7 +274,132 @@ mod tests {
                 scaling_exponents: vec![0, 0, 12],
                 block_created: 1337,
             }
-        )
+        );
+    }
+
+    #[tokio::test]
+    async fn fetch_pool_state() {
+        let pool_id = H256([0x90; 32]);
+        let tokens = [H160([1; 20]), H160([2; 20]), H160([3; 20])];
+        let balances = [bfp!("1000.0"), bfp!("10.0"), bfp!("15.0")];
```

Comment:
cool macro... was this the only reason for introducing it? Is this easier than `bfp::from`?

Reply:
Hmm good question! Didn't know we had a `From` conversion for strings to Balancer fixed point numbers.

I had seen `"...".parse::<Bfp>().unwrap()` before.

Reply:
I checked and indeed there is no `From` conversion for strings implemented (which makes sense because it can fail).